# Importing the libraries

In [21]:
import pandas as pd
import numpy as np

In [15]:
! conda install -c plotly plotly 

Solving environment: done

## Package Plan ##

  environment location: /home/moi/anaconda3/envs/myenv

  added / updated specs:
    - plotly


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    retrying-1.3.3             |           py37_2          16 KB
    ------------------------------------------------------------
                                           Total:          16 KB

The following NEW packages will be INSTALLED:

  plotly             plotly/noarch::plotly-4.12.0-py_0
  retrying           pkgs/main/linux-64::retrying-1.3.3-py37_2


Proceed ([y]/n)? ^C

CondaSystemExit: 
Operation aborted.  Exiting.



In [2]:
import plotly.express as px

df = px.data.gapminder().query("country=='Canada'")
fig = px.line(df, x="year", y="lifeExp", title='Life expectancy in Canada')
fig.show()

ModuleNotFoundError: No module named 'plotly'

In [3]:
df = pd.read_csv("proteinGroups.csv", low_memory=False)
df.head(10)

Protein IDs Majority protein IDs Peptide counts (all)  \
0  A0A024RBG1           A0A024RBG1                    7   
1  A0A096LP55               P07919           A0A096LP55   
2      P0DPI2           A0A0B4J2D5               P0DPI2   
3      P0DPI3           A0A0U1RR11           A0A0U1RRI6   
4      A0AV96               Q8TBY0               A0AV96   
5      A0AVF1               A0AVF1                    5   
6      A0AVT1               A0AVT1                   35   
7      A0FGR8               A0FGR8                   24   
8      A0JLT2               A0JLT2                    3   
9      A0JNW5               A0JNW5                    3   

  Peptide counts (razor+unique) Peptide counts (unique)  \
0                             2                       2   
1                        P07919                       1   
2                    A0A0B4J2D5                       7   
3                        P0DPI3              A0A0U1RR11   
4                            13                       1   
5                             5                       5   
6                            35                      35   
7                            24                      24   
8                             3                       3   
9                             3                       3   

                                       Protein names Gene names Fasta headers  \
0                                                NaN      NUDT4           NaN   
1                                                  1          1             1   
2                                                  7          7             7   
3                                         A0A0U1RRI6          1             1   
4                                                 13          1            12   
5                Intraflagellar transport protein 56      TTC26           NaN   
6        Ubiquitin-like modifier-activating enzyme 6       UBA6           NaN   
7                           Extended synaptotagmin-2      ESYT2           NaN   
8  Mediator of RNA polymerase II transcription su...      MED19           NaN   
9                       UHRF1-binding protein 1-like  UHRF1BP1L           NaN   

  Number of proteins                Peptides  ... Unnamed: 1014 Unnamed: 1015  \
0                  1                       7  ...           NaN           NaN   
1                  1                       1  ...           NaN           NaN   
2                  7                       7  ...           NaN           NaN   
3                  1                       1  ...           NaN           NaN   
4                  0  RNA-binding protein 47  ...           NaN           NaN   
5                  1                       5  ...           NaN           NaN   
6                  1                      35  ...           NaN           NaN   
7                  1                      24  ...           NaN           NaN   
8                  1                       3  ...           NaN           NaN   
9                  1                       3  ...           NaN           NaN   

  Unnamed: 1016 Unnamed: 1017 Unnamed: 1018 Unnamed: 1019 Unnamed: 1020  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   
2           NaN           NaN           NaN           NaN           NaN   
3           NaN           NaN           NaN           NaN           NaN   
4           NaN           NaN           NaN           NaN           NaN   
5           NaN           NaN           NaN           NaN           NaN   
6           NaN           NaN           NaN           NaN           NaN   
7           NaN           NaN           NaN           NaN           NaN   
8           NaN           NaN           NaN           NaN           NaN   
9           NaN           NaN           NaN           NaN           NaN   

  Unnamed: 1021 Unnamed: 1022 Unnamed: 1023  
0           NaN           NaN           NaN  


# How many proteins were detected in total?

In [4]:
df['Protein IDs'].nunique()

7499

In [6]:
df2 = df[~df['Protein IDs'].str.startswith('CON')]
df3 = df2[~df2['Protein IDs'].str.startswith('REV')]
df3['Protein IDs'].nunique()

7251

In [8]:
df3['Protein IDs'].value_counts().sum()

7251

In [9]:
df3.shape

(7251, 1024)

In [16]:
df3.to_csv("my-sample.csv",index_label=False)

In [10]:
df4 = df3['Protein IDs'].dropna(axis=0)

In [11]:
df4.shape

(7251,)

So, total number of unique protein presents are 7251